src/deterministic_vi/elbo_kl.jl

```diff
@@ -40,23 +40,21 @@ function gen_beta_kl{NumType <: Number}(alpha2::NumType, beta2::NumType)
         grad = zeros(NumType2, 2)
         hess = zeros(NumType2, 2, 2)
 
-        if calculate_derivs
-            trigamma_alpha1 = trigamma(alpha1)
-            trigamma_beta1 = trigamma(beta1)
-            trigamma_both = trigamma(alpha1 + beta1)
-            grad[1] = alpha_diff * trigamma_alpha1 + both_inv_diff * trigamma_both
-            grad[2] = beta_diff * trigamma_beta1 + both_inv_diff * trigamma_both
-
-            quadgamma_both = polygamma(2, alpha1 + beta1)
-            hess[1, 1] = alpha_diff * polygamma(2, alpha1) +
-                         both_inv_diff * quadgamma_both +
-                         trigamma_alpha1 - trigamma_both
-            hess[2, 2] = beta_diff * polygamma(2, beta1) +
-                         both_inv_diff * quadgamma_both +
-                         trigamma_beta1 - trigamma_both
-            hess[1, 2] = hess[2, 1] =
-                -trigamma_both + both_inv_diff * quadgamma_both
-        end
+        trigamma_alpha1 = trigamma(alpha1)
```

Comment:
That was my thinking. @jrevels's implementation with auto diff may be more memory/computation efficient.